Create erb wizard from

iBrooks opened the pull request:
Added an erb form to add a new wizard
removed obsolete api endpoints
removed obsolete api tests
updated show to operate off of id not name

Commit:
without NavBar

Commit:
with NavBar

Commit:
Merge branch 'master' of https://github.com/prophetsfall/wizards_reviews into wizard-reviews

Commit:
created reviews

alcodes299 committed:
added erb form for wizards

iBrooks committed:
things

iBrooks committed:
Merge remote-tracking branch 'origin/wizard-reviews' into create-erb-wizard-from

iBrooks committed:
problems abound

iBrooks committed:
added to_param to wizard model
added erb form for new wizard
reconfigured endpoint
added create to WizardsController (not api)

iBrooks committed:
Merge branch 'master' into create-erb-wizard-from

iBrooks committed:
added rails-env

iBrooks committed:
Merge branch 'master' into create-erb-wizard-from

iBrooks committed:
removed obsolete tests
updated controller to operate off of id not name
re-enabled pending tests and fixed to pass

dpickett commented on app/controllers/api/v1/wizards_controller.rb:
```diff
@@ -24,6 +24,6 @@ def show
 
   def wizard_params
-    params.require(:wizard).permit(:name, :description, :img_url, :creator_id)
+    params.require(:wizard).permit(:id, :name, :description, :img_url, :creator_id)
```
we do not want to allow the user to provide the id / primary key here. this should be left out of strong params

dpickett commented on app/controllers/wizards_controller.rb:
```diff
@@ -2,4 +2,34 @@ class WizardsController < ApplicationController
+  def show
+    render :index
+  end
+  def new
```
consider a blank line between each method

dpickett commented on app/controllers/wizards_controller.rb:
```diff
@@ -2,4 +2,34 @@ class WizardsController < ApplicationController
+    if current_user
+      if @wizard.save
+        redirect_to wizard_path(@wizard)
+        flash[:notice] = 'Wizards added successfully'
```
convention wise, we like the `flash` to come before the redirect.

dpickett commented on app/javascript/react/components/WizardShow.js:
```diff
@@ -1,13 +1,21 @@
     <p>{props.description}</p>
   </div>
+
+  <div>
```
consider indenting this `<div>` for readability

dpickett commented on app/models/wizard.rb:
```diff
@@ -3,4 +3,7 @@ class Wizard < ApplicationRecord
 
   validates :name, presence: true, uniqueness: true
   validates :description, presence: true
+  def to_param
```
consider a line break above this line

dpickett commented on db/migrate/20180124144439_add_name_too_users.rb:
```diff
@@ -1,5 +1,5 @@
 class AddNameTooUsers < ActiveRecord::Migration[5.1]
   def change
-    add_column :users, :user_name, :string, null: false
+    add_column :users, :user_name, :string, null: false, default: "user"
```
why set a default here? I wouldn't recommend it.

dpickett commented on spec/features/user_sumbits_new_wizard_spec.rb:
```diff
@@ -0,0 +1,29 @@
+
+    expect(page).to have_content 'Wizards added successfully'
+  end
+  scenario 'user submits invalid form' do
```
these tests should be completed

iBrooks committed:
Actually removed deprecated api endpoints

added whitespace
fixed indents
reordered flash msg
the tests were already complete I DONT KNOW WHAT YOURE TALKING ABOUT
created migration to remove default for user (PAT D: )

iBrooks committed:
rake db:test:prepare

iBrooks committed:
Merge remote-tracking branch 'origin/master' into create-erb-wizard-from

opted to set the api controller tests as pending in case we opt to
refactor to react during stretch goals

prophetsfall committed:
cleaned up extra controller actions

prophetsfall committed:
Merge branch 'master' into create-erb-wizard-from

prophetsfall committed:
Merge branch 'create-erb-wizard-from' of https://github.com/prophetsfall/wizards_reviews into create-erb-wizard-from

prophetsfall committed:
corrected routing error in wizard_controller.

prophetsfall commented:
We've cleaned this up pull request. Please review.

dpickett commented on app/controllers/wizards_controller.rb:
```diff
@@ -2,19 +2,22 @@ class WizardsController < ApplicationController
   def create
-    @wizard = Wizard.new(wizards_params)
     if current_user
+      @wizard = Wizard.new(wizards_params.merge({ creator_id: current_user.id }))
```
I would prefer:

```ruby
@wizard = Wizard.new(wizards_params)
@wizard.user = current_user
```

dpickett commented on app/views/wizards/new.html.erb:
```diff
@@ -1,8 +1,8 @@
  <% if @wizard.errors.any? %>
- <div id=“error_explanation”>
-   <h2><%= pluralize(@wizard.errors.count, “error”) %> prohibited this Wizard from being saved:</h2>
+ <div id='error_explanation'>
```
use double quotes for HTML attribute values

prophetsfall committed:
Updated quotes on new template, changed how we set the creator on new Wizard, cleaned up some pluralization.

prophetsfall committed:
fixed index test

prophetsfall commented:
Requested changes have been completed.

prophetsfall committed:
removed some white space.

prophetsfall committed:
Merge branch 'master' of https://github.com/prophetsfall/wizards_reviews into create-erb-wizard-from

prophetsfall commented:
Ok, this should be good to go now.

dpickett commented on app/controllers/wizards_controller.rb:
```diff
@@ -17,12 +17,13 @@ def new
     if current_user
-      @wizard = Wizard.new(wizards_params.merge({ creator_id: current_user.id }))
+      @wizard = Wizard.new(wizards_params)
+      @wizard.creator_id = current_user.id
```
you could remove the id here - I find it to be more expressive

`@wizard.creator = current-user`
